contribs/gmf/src/services/datasourcesmanager.js

```diff
@@ -579,6 +579,29 @@ gmf.DataSourcesManager = class {
     goog.asserts.assert(dataSource, 'DataSource should be set');
     const visible = newVal === 'on';
     dataSource.visible = visible;
+
+    // In GMF, multiple data sources can be combined into one ol.layer.Layer
+    // object. When changing the state of a data source, we need to make
+    // sure that the FILTER param match order of the current LAYERS param.
+    //
+    // Note: we only need to do this ONCE, as there can be only one
+    // data source being filtered at a time
+    const siblingDataSourceIds = gmf.SyncLayertreeMap.getLayer(
+      treeCtrl).get('querySourceIds');
+    if (Array.isArray(siblingDataSourceIds)) {
+      const ngeoDataSources = this.ngeoDataSources_.getArray();
+      for (const ngeoDataSource of ngeoDataSources) {
+        if (ol.array.includes(siblingDataSourceIds, ngeoDataSource.id) &&
```

Comment:
I disagree.  Closure Compiler is very smart.  Most of the time, for simple methods such as the above, while compressing, it even replaces the call of the method entirely.

OK, maybe `ol.array.includes` is a bit far fetched, i.e. it's one line of code to do the equivalent in pure JavaScript, so I kinda agree in a way, but at the same time the syntax is easier with the method and it serves a purpose: be an util method.

So, because we know the Compiler is smart and also I think `ol.array.includes` is a good utility method (otherwise, why would have it been written in the first place) the I suggest we keep it.

Reply:
The `ol.array.includes` method exists to backport ES6 functionality for projects coded in ES5. It is not the case anymore for us in ngeo where we use ES6 and our build chain already handles the transformation for us (I checked the generated code).

Currently in ngeo, with have both code using the deprecated `ol.array.includes` method and the standard. I don't

Reply:
I understand.  I do not have the "reflexes" to "think with ES6 in mind" but you're right, ngeo wants to use ES6.  So, I assume that through the use of polyfills this method should work in IE, otherwise it won't: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

I get it.  Thanks for the explanations.

Reply:
Yes.

If you check the generated code, you can see the polyfills are injected like these:
```js
wa("Array.prototype.includes",function(a){return a?a:function(a,c){var b=this;b instanceof String&&(b=String(b));var e=b.length;for(c=c||0;c<e;c++)if(b[c]==a||Object.is(b[c],a))return!0;return!1}});
wa("String.prototype.includes",function(a){return a?a:function(a,c){if(null==this)throw new TypeError("The 'this' value for String.prototype.includes must not be null or undefined");if(a instanceof RegExp)throw new TypeError("First argument to String.prototype.in
wa("Array.prototype.find",function(a){return a?a
```

Reply:
Yep, that makes sense.  Thanks.